docs/docs/api-reference/core/Snapshot.md

```diff
@@ -0,0 +1,58 @@
+---
+title: class Snapshot
+sidebar_label: Snapshot
+---
+
+A `Snapshot` object represents an immutable snapshot of the state of Recoil [atoms](/docs/api-reference/core/atom).  It is inteded to standardize the API for observing, inspecting, and managing global Reocil state.  It is mostly useful for dev tools, global state synchronization, history navigation, &c.
+
+```jsx
+class Snapshot {
+  getLoadable: <T>(RecoilValue<T>) => Loadable<T>;
+  getPromise: <T>(RecoilValue<T>) => Promise<T>;
+
+  map: (MutableSnapshot => void) => Snapshot;
+  asyncMap: (MutableSnapshot => Promise<void>) => Promise<Snapshot>;
+}
+```
+
+### Reading Snapshots
+
+Snapshots are read-only with respect to atom state.  They can be used to read atom state and evaluate selectors derived state.  The `getPromise()` method can be used to wait for the evaluated value of asynchronous selectors, so you can see what the value would be based on the static atom state.
```

Comment:
grammar nit: change "and evaluate selectors derived state" to:

"and evaluate selectors' derived state"
"and evaluate derived state from selectors"
"and evaluate a selector's derived state"